libraries/AP_HAL_ChibiOS/Scheduler.cpp

```diff
@@ -333,6 +333,10 @@ void Scheduler::_timer_thread(void *arg)
 */
 bool Scheduler::in_expected_delay(void) const
 {
+    if (!_initialized) {
```

Comment:
Perhaps this should be `_hal_initialised` instead.  That was we're not exempting the early *vehicle* setup code from the checks.

https://github.com/ardupilot/ardupilot/blob/master/libraries/AP_HAL_ChibiOS/HAL_ChibiOS_Class.cpp#L211

Reply:
no, we don't want delays in setup() to generate errors, that is the point of this change. During setup() we expect timing to be weird. Same reason we don't start the watchdog till after setup() completes